src/icp_v2.cc

```diff
@@ -680,16 +662,26 @@ icpHandleUdp(int sock, void *)
         debugs(12, 4, "icpHandleUdp: FD " << sock << ": received " <<
                (unsigned long int)len << " bytes from " << from);
 
-#ifdef ICP_PACKET_DUMP
-
-        icpPktDump(buf);
-#endif
-
         if ((size_t) len < sizeof(icp_common_t)) {
             debugs(12, 4, "icpHandleUdp: Ignoring too-small UDP packet");
             break;
         }
 
+        auto pkt = reinterpret_cast<icp_common_t *>(buf);
+        struct in_addr a;
+        a.s_addr = pkt->shostid;
+        debugs(12, 2, "ICP Client remote=" << from << " FD " << sock);
+        debugs(12, 2, "ICP Client REQUEST:\n---------\n" <<
+               "opcode=(" << std::setw(3) << pkt->opcode << ")" << icp_opcode_str[pkt->opcode] <<
+               ", version=" << pkt->version <<
+               ", length=" << ntohs(pkt->length) <<
+               ", reqnum=" << std::setw(8) << ntohl(pkt->reqnum) <<
+               ", flags=(" << std::hex << std::setw(8) << ntohl(pkt->flags) << ")" <<
```

Comment:
Here and elsewhere, please do not attempt to align/beautify level-2+ debugging output like that. It wastes runtime CPU cycles, log space, code space, compile time, developer time, and often complicates automated log processing. The tiny benefits of using setw() do not outweigh the many cons. Please also avoid excessive decorations like parenthesis around numbers, for similar reasons.

```suggestion
               ", reqnum=" << ntohl(pkt->reqnum) <<
               ", flags=" << std::hex << ntohl(pkt->flags) << std::dec <<
```

The above suggestion also removes "std::hex" long-term side effects.